jax/lax/lax_control_flow.py

```diff
@@ -744,6 +826,36 @@ def f_aug(*args):
 
   return _make_typed_jaxpr(f_aug, jaxpr.in_avals)
 
+def _join_cond_pe_staged_jaxpr_inputs(jaxprs, res_avals_per_jaxpr):
+  newvar = core.gensym('~')     # TODO(frostig): safer gensym
```

Comment:
Maybe you can pass the newvar in from the caller.

Reply:
I think we should fix the root issue and make a generally safer gensym (given a jaxpr), then use it here. The TODO is meant as a note to that end.

Reply:
Resolved by #3211